utils/reconciler.go

```diff
@@ -338,32 +362,21 @@ func (r *ReconcilerBase) ManageSuccess(conditionType common.StatusConditionType,
 
 	var retryInterval time.Duration
 
-	// If the resources are not ready
-	if newCondition.GetStatus() != corev1.ConditionTrue {
-		// If the resources were ready and now they are not, or encountered a different error
+	// If the application or resources are not ready
+	if oldCondition == nil || oldCondition.GetStatus() != newCondition.GetStatus() || oldCondition.GetMessage() != newCondition.GetMessage() || oldRecCondition == nil || oldRecCondition.GetStatus() != newRecCondition.GetStatus() || oldRecCondition.GetMessage() != newRecCondition.GetMessage() {
```

Comment:
Just wondering whether `oldCondition.GetReason() != newCondition.GetReason()` is needed here as well?